src/ngraph/runtime/cpu/cpu_aligned_buffer.hpp

```diff
@@ -0,0 +1,56 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cstddef>
+#include "ngraph/runtime/cpu/cpu_allocator.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace cpu
+        {
+            class CPUAlignedBuffer;
+        }
+    }
+}
+
+/// \brief Allocates a block of memory on the specified alignment. The actual size of the
+/// allocated memory is larger than the requested size by the alignment, so allocating 1 byte
+/// on 64 byte alignment will allocate 65 bytes.
+class ngraph::runtime::cpu::CPUAlignedBuffer
```

Comment:
This looks just like runtime::AlignedBuffer. Why not just use that class and add an optional `allocator` pointer that defaults to nullptr? It will save code duplication.